crates/net/network/src/manager.rs

```diff
@@ -899,27 +899,3 @@ pub enum NetworkEvent {
     /// Event emitted when a new peer is removed
     PeerRemoved(PeerId),
 }
-
-/// Bundles all listeners for [`NetworkEvent`]s.
-#[derive(Default)]
-struct NetworkEventListeners {
-    /// All listeners for an event
-    listeners: Vec<mpsc::UnboundedSender<NetworkEvent>>,
-}
-
-// === impl NetworkEventListeners ===
-
-impl NetworkEventListeners {
-    /// Sends  the event to all listeners.
-    ///
-    /// Remove channels that got closed.
-    fn send(&mut self, event: NetworkEvent) {
-        self.listeners.retain(|listener| {
-            let open = listener.send(event.clone()).is_ok();
-            if !open {
-                trace!(target : "net", "event listener channel closed",);
-            }
```

Comment:
do we think this trace is useful / worth keeping in the new impl?

Reply:
yeh, doesn't hurt imo